app/code/Magento/MediaGallery/Model/DataExtractor.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallery\Model;
+
+use Magento\MediaGalleryApi\Model\DataExtractorInterface;
+
+/**
+ * Extract data from an object using available getters
+ */
+class DataExtractor implements DataExtractorInterface
+{
+    /**
+     * @inheritdoc
+     */
+    public function extract($object, string $interface = null): array
+    {
+        $data = [];
+
+        $reflectionClass = new \ReflectionClass($interface ?? $object);
```

Comment:
As I understand this extractor is used to extract data from `AssetInterface` during save (this is the only use case I could find in the code).
It looks for me overcomplicated.
Would it be better to just remove `AssetInterface` (is it really necessary?) and admit that we use object of `Asset` class. It is clear that this object is an ancestor of `DataObject`, which has clear `getData()` method. So we could just call this method to get all necessary data. This looks more clear:
* the code would be more straightforward (due to fewer abstractions)
* we rely on clear interface (`getData()` method) instead of guessing which methods we should call to retrieve necessary data

Reply:
Or another approach - use interface's getters explicitly.

Reply:
Hello @buskamuza ,

My understanding of benefits provided from this approach is: we have a tool which allows as to get only object properties declared in the contract. It is useful when you prepare your object for save/update or clone during the work with it. You can rely on it and be sure that you will get precisely the declared object values and nothing more. In case when we use the `getData()` method, we will need to write additional code which will exclude not object data if it exists. So, data extractor automates only the contract declared object properties data extraction, which makes it more consistent.

Reply:
The extractor can be used for extracting data from any data interface (replacement of entity manager hydrator). It is currently used for extracting data from other entities in the Adobe Stock Integration package. 

AssetInterface is the interface designed to describe the asset entity. The interface contains getters necessary to retrieve the asset information. `AssetInterface` does not include `getData` method, so I don't think we can safely rely on it.

Reply:
In addition to guys' comments, from my point of view, it looks strange to have getters and `getData()` in the same interface.